From 0a627f7f8ed52f44bf73f4a7bcf8b1ca4ce95411 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Fri, 28 Jun 2024 11:28:09 +0800
Subject: [PATCH 6/9] lower EXT_texture_norm16 to warning.

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 libobs-opengl/gl-subsystem.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/libobs-opengl/gl-subsystem.c b/libobs-opengl/gl-subsystem.c
index 5afb82316..d2419a40b 100644
--- a/libobs-opengl/gl-subsystem.c
+++ b/libobs-opengl/gl-subsystem.c
@@ -154,9 +154,8 @@ static bool gl_init_extensions(struct gs_device *device)
 	}
 
 	if (!GLAD_GL_EXT_texture_norm16) {
-		blog(LOG_ERROR, "OpenGL ES extension EXT_texture_norm16 "
-				"is required.");
-		return false;
+		blog(LOG_WARNING, "OpenGL ES extension EXT_texture_norm16 "
+				  "is recommended but missing, expect functionality loss.");
 	}
 
 	if (!GLAD_GL_EXT_disjoint_timer_query) {
-- 
2.40.1

